.github: chown workspace for render_test_results

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62207

Workspace was getting held back due to permission denied errors, let's
ensure we have a chown'd / clean workspace for all render_test_results
runs

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Should reduce the amount of errors found like:

<img width="1202" alt="Screen Shot 2021-07-26 at 11 29 07 AM" src="https://user-images.githubusercontent.com/1700823/127040008-76459c97-a496-4a30-a169-483553f0d6e3.png">

see: https://github.com/pytorch/pytorch/runs/3164470654

Differential Revision: [D29915232](https://our.internmc.facebook.com/intern/diff/D29915232)

seemethere committed:
.github: chown workspace for render_test_results

Workspace was getting held back due to permission denied errors, let's
ensure we have a chown'd / clean workspace for all render_test_results
runs

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29915232).

zhouzhuojie reviewed (approved):
🎉  Also, attached to the dev infra project.

malfet commented on .github/templates/linux_ci_workflow.yml.j2:
```diff
@@ -405,6 +405,13 @@ jobs:
+          docker run --rm -v "$(pwd)/../":/v -w /v "${ALPINE_IMAGE}" chown -R "$(id -u):$(id -g)" .
+      - name: Clean workspace
+        run: |
+          rm -rf "${GITHUB_WORKSPACE:?}/*"
```
This wouldn't removes files/folders starting with dot, would it?

walterddr replied:
maybe simply this?
```suggestion
          rm -rf "${GITHUB_WORKSPACE:?}/"
```

seemethere replied:
This will error out though unless we recreate the `GITHUB_WORKSPACE`, this `rm -rf` here is just for safety since the chown above should be the thing that actually fixes permissions on the `checkout` 

I also believe `actions/checkout` does a `rm -rf` of the space as well, hence we received the permission denied error on `actions/checkout`

seemethere committed:
Update on ".github: chown workspace for render_test_results"


Workspace was getting held back due to permission denied errors, let's
ensure we have a chown'd / clean workspace for all render_test_results
runs

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Should reduce the amount of errors found like:

<img width="1202" alt="Screen Shot 2021-07-26 at 11 29 07 AM" src="https://user-images.githubusercontent.com/1700823/127040008-76459c97-a496-4a30-a169-483553f0d6e3.png">

see: https://github.com/pytorch/pytorch/runs/3164470654

Differential Revision: [D29915232](https://our.internmc.facebook.com/intern/diff/D29915232)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29915232).
